From: Jiri Slaby <jslaby@suse.cz>
Subject: fix for kernel 6.14 build
References: kernel 6.14

For these kernel commits:
5be1fa8abd7b Pass parent directory inode and expected name to ->d_revalidate()
cb2e1c2136f7 drm: remove driver date from struct drm_driver and all drivers

---
 src/VBox/Additions/linux/drm/vbox_drv.c        |    2 ++
 src/VBox/Additions/linux/sharedfolders/utils.c |    6 +++++-
 2 files changed, 7 insertions(+), 1 deletion(-)

--- a/src/VBox/Additions/linux/drm/vbox_drv.c
+++ b/src/VBox/Additions/linux/drm/vbox_drv.c
@@ -376,7 +376,9 @@ static struct drm_driver driver = {
 #endif
 	.name = DRIVER_NAME,
 	.desc = DRIVER_DESC,
+#if RTLNX_VER_MAX(6,14,0)
 	.date = DRIVER_DATE,
+#endif
 	.major = DRIVER_MAJOR,
 	.minor = DRIVER_MINOR,
 	.patchlevel = DRIVER_PATCHLEVEL,
--- a/src/VBox/Additions/linux/sharedfolders/utils.c
+++ b/src/VBox/Additions/linux/sharedfolders/utils.c
@@ -1162,7 +1162,11 @@ int vbsf_path_from_dentry(struct vbsf_su
  *
  * @note Caller holds no relevant locks, just a dentry reference.
  */
-#if RTLNX_VER_MIN(3,6,0)
+#if RTLNX_VER_MIN(6,14,0)
+static int vbsf_dentry_revalidate(struct inode *parent_dir,
+				  const struct qstr *name,
+				  struct dentry *dentry, unsigned flags)
+#elif RTLNX_VER_MIN(3,6,0)
 static int vbsf_dentry_revalidate(struct dentry *dentry, unsigned flags)
 #elif RTLNX_VER_MIN(2,6,0)
 static int vbsf_dentry_revalidate(struct dentry *dentry, struct nameidata *nd)
